Fixes for ansible-console usability.

ylmrx opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
- Fixes #68529 : changed Ctrl+C and Ctrl+D , to something closer to a standard shell behavior. (ie. cancel/reset current line, end session instead of aborting current session)
- Fixes #68529 : added an exit message
- #72195 : add more documentation, describing the various commands
- Few typos
- Add more information on configuration transition: it should clearly state it's switching to/from Check mode even on quieter verbose modes for example

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ansible-console

bcoca reviewed (changes requested):
needs a changelog, also not sure about the change in Ctrl+C, might need a toggle and deprecation period

ylmrx committed:
Fixes for ansible-console usability

 - Fix Ctrl+C/D behavior (#68529)
 - correct few typos and missing infos
 - Fixes doc (#72195)
